Sources/_StringProcessing/Unicode/CaseConversion.swift

```diff
@@ -9,6 +9,24 @@
 //
 //===----------------------------------------------------------------------===//
 
+@_spi(_Unicode) import Swift
 
 // TODO
 
+extension Character {
+  /// Whether this character and `c` are equal when case folded.
+  func caseFoldedEquals(_ c: Character) -> Bool {
+    guard #available(SwiftStdlib 5.7, *) else { fatalError() }
+    let foldedSelf = unicodeScalars.map(\.properties._caseFolded).joined()
+    let foldedOther = c.unicodeScalars.map(\.properties._caseFolded).joined()
+    return foldedSelf == foldedOther
```

Comment:
This makes a lot of intermediary arrays, which is unfortunate because nearly all Characters fit in the small-string form, and those that don't are usually case invariant. Does `.lazy.map(...).elementsEqual(` work?

Reply:
I think what we really want is a stdlib API that can do the case-folded comparison — it's wasteful even to UTF-8 encode these when we should be able to convert, canonicalize, and compare character-by-character.

Reply:
We can write the code here or in the stdlib (I agree stdlib SPI makes sense). But in the meantime, can we write the code we want the stdlib to have, or at least approximate it by making the mapping lazy?

Reply:
Otherwise, it looks like case-insensitive matching could do multiple array allocations for every single character in the input